### ISSUE TITLE
Snotify Async - Adding Config Options Makes 2 Async Calls

### ISSUE THREAD

**Issue by Mornney** [artemsky/vue-snotify#54]
Hi, 

I did create an issue about Snotify not working correctly with Select2, but I believe this is not the case, Select2 was refreshing the drop downs due to a 2nd call being made in the Async. 

From what I can tell if I add extra config values into the config option of the resolve / reject this causes the async call to be done twice instead of once.

Here is the example resolve Async from the documentation with a few added config values into the resolve config option, if you run this up you will see that the console log ("calling1") ill be shown twice in the console.

```
this.$snotify.async('Called with promise', 'Success async', () => new Promise((resolve, reject) => {
            console.log('calling1');
            setTimeout(() => resolve({
              title: 'Success!!!',
              body: 'We got an example success!',
              config: {
                closeOnClick: true,
                position: "rightTop",
                backdrop: 0.5,
                timeout: 200
              }
            }), 2000);
          }));
```

Is there anything I can do to change this?

basically I want to have the Async appear in the middle of the screen, then I want the resolve "success" popup to show in the top right hand corner, doing this however causes this double axios call issue. 

Thanks
Peter

**Comment by javierrivarola** [comment]
**I'm having the same issue, this is my code:** 

`this.$snotify.async('Borrando dispositivo', 'Esta accion es permanente.', () => new Promise((resolve, reject) => {
                                    console.log("calling async")
                                    axios.delete('/devices/' + device.id)
                                        .then(response => {
                                            console.log("got response async", response)

                                            if (response.data.success == true) { //In case if data is deleted
                                                console.log("response data is success true")

                                                resolve({
                                                    title: 'Listo!',
                                                    body: 'El dispositivo se borro de tu cuenta.',
                                                    config: {
                                                        closeOnClick: true,
                                                        timeout: 500,

                                                    }
                                                })
                                                this.devices.splice(this.devices.indexOf(device), 1);

                                            } else {
                                                console.log("response data is success false")

                                                reject({
                                                    title: 'Error!!!',
                                                    body: response.data.data,
                                                    config: {
                                                        closeOnClick: true,
                                                        position: "rightTop",

                                                    }
                                                })
                                            }
                                        }).catch(error => {
                                        reject({
                                            title: 'No se pudo borrar el dispositivo!',
                                            body: 'Verifica que estes conectado a internet.',
                                            config: {
                                                closeOnClick: true,
                                                timeout: 1000,

                                            }
                                        })
                                        console.log(error);
                                    });


                            })

                            );`

**Comment by nukkas** [comment]
We are having the same issue 
<img width="1269" alt="Screen Shot 2019-08-29 at 16 22 53" src="https://user-images.githubusercontent.com/52958615/63934695-8bfd5600-ca79-11e9-8cc2-e20958b55b32.png">


this.$snotify.async(
          "Mode Premium Amount",
          "Calculating...",

          () =>
            new Promise((resolve, reject) => {
              return this.getChangeDataResponse()
                .then(function(response) {
                  if (response.status && response.type == undefined) {
                    resolve({
                      title: response.error,
                      body: "Mode Premium Amount",
                      config: {
                        closeOnClick: true,
                        position: SnotifyPosition.centerTop
                      }
                    });
                  } else {
                    reject({
                      title: "Error",
                      body: response.error,
                      config: {
                        closeOnClick: true,
                        position: SnotifyPosition.centerTop,
                      }
                    });
                  }
                })
                .catch(function(error) {});
            })
        );

**Comment by jonascarlbaum** [comment]
I read your comments, and although not pretty, this solved it for me...

```
var promise = null;

this.$snotify.async('Message...', 'Header', () => {
    if (!!promise) {
        return promise;
    }
    else {
        promise = new Promise((resolve, reject) => {
            ...api-calls...resolve...reject...etc.
        });
        return promise;
    }
});
```

**Comment by michaellanpart** [comment]
I was able to duplicate this issue. 
**ISSUE**: If you are creating components that are used as child components where both the parent and the child component have the `<vue-snotify></vue-snotify>` tag, you will see multiple network events happen while using the $snotify.async. 
**FIX**: I have not found an elegant fix yet. However, what has worked is in the event  I have several components that are loaded I remove the <vue-snotify></vue-snotify> tag on all except one. That seems to fix the issue.  This is a poor fix because these are components and if I use one outside the designed method it should still work and by removing the tag it will not. 
**QUESTION**: Is this happening because I am using snotify incorrectly, by putting the `<vue-snotify></vue-snotify>` tag in every component where I use snotify?    

See final FIX below in my post.

**Comment by jonascarlbaum** [comment]
> I was able to duplicate this issue.
> **ISSUE**: If you are creating components that are used as child components where both the parent and the child component have the "" tag, you will see multiple network events happen while using the $snotify.async.
> **FIX**: I have not found an elegant fix yet. However, what has worked is in the event I have several components that are loaded I remove the tag on all except one. That seems to fix the issue. This is a poor fix because these are components and if I use one outside the designed method it should still work and by removing the tag it will not.
> **QUESTION**: Is this happening because I am using snotify incorrectly, by putting the tag in every component where I use snotify?

Interesting catch. I also use the `<snotify>`-tag in each component that I need to output messages, since I don’t know any better way... I’m not having a SPA either. Components are self contained and should be able to output `snotify`-messages no matter what parent they have…

Your catch leads me to say always put your promise as a const/var and return it instead of creating promises for self and each parent component too, almost as my suggestion above.

I think I’ve seen multiple outputs with my solution too, it’s strange when debugging I didn’t see multiple hits in code, which indicates there are multiple components running on the same promise in my case, so I kind of can confirm your catch in and assumptions.

Ideally it’s solved within the library, everything else is workarounds…

**Comment by michaellanpart** [comment]
Ok, I found a fix for my situation.  
**ENVIRONMENT**: If you are using a SPA (Single Page Application) environment and are placing the tag `<vue-snotify></vue-snotify>` in each component.  

**REPRODUCE**: When this.$snotify.async is used and if you have several components with the ` <vue-snotify></vue-snotify>` tag in them while nested components you will see duplicate ajax, or in my case axios calls.  

**FIX**: Remove all ` <vue-snotify></vue-snotify>` tags from your components and place one tag in the Container file. Here is where I placed mine. This works!  
![image](https://user-images.githubusercontent.com/60103259/97209474-64437400-178a-11eb-9970-f94227db501a.png)

**Comment by jonascarlbaum** [comment]
Yeah, I managed to rework my non-SPA components to have only one `<vue-snotify>`. So all rendered master-components have these, and not other components. It works, it was wrong usage...
